### PR TITLE
Decorate class.

### PR DESCRIPTION
**Pull request by Muhammad0602**
### For this milestone of the project I have implemented the following:
- Added a new class `Nameable`.
- Turned my `Person` class to `Nameable`.
- Prepared base `Decorator`.
- Prepared `CapitalizeDecorator` and `TrimmerDecorator` classes based on the `Decorator`.

### CODE REVIEW

File: capitalize_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './decorator'
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
+  end
+end
```
Comment (omar-muhamad) [inline capitalize_decorator.rb]: - [ ] The capitalize decorator suppose to capitalize the first character of the name not making the whole name in uppercase, so please fix that.
[Hint] you can do it like that:
```ruby
 def correct_name
    @nameable.correct_name.capitalize
  end
```

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Muhammad0602,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: capitalize_decorator.rb (hunk 1 above)
Comment (Muhammad0602) [reply capitalize_decorator.rb]: Hello @Omar-Muhamad, thank you for the feedback. I will do it immediately.

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🥳

Hi @Muhammad0602 🙋‍♀️

Great work implementing the changes suggested by the previous reviewer 💪**_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Class is decorated well, perfect!
-  ✅ Great work using the right Gitflow 
-  ✅ Your README looks professional, well done!
-  ✅ Commits are descriptive, good work.
-  ✅ Linters are working properly, great job! 

## Optional suggestions ❕
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Muhammad0602** [comment]
Thanks for the review @KanzaTahreem.

**Pull request merged**